tests/testthat/test-coerce_date.R

```diff
@@ -12,6 +12,13 @@ test_that("coerce_date works for non-garbage in", {
   expect_error(
     coerce_date(NULL), NA
   )
+  expect_error(
+    coerce_date(NULL), NA
+  )
+  expect_identical(
+    coerce_date(NULL),
+    data.table::as.IDate(.Date(numeric()))
+  )
```

Comment:
bit confused as to how both these tests can pass. also, ibid previous suggestion on simplifying the arg to as.IDate

Reply:
i mean, both expect_error (which if it passes, means coerce_date(NULL) shouldn't return anything) and expect_identical (which should only pass if something returned).

Reply:
It's because the NA in expect_error makes it expect no error I think. Not ideal syntax.

Reply:
Then it still seems like the expect_identical is sufficient - that will definitely not succeed if the NULL argument returns an error.

Reply:
I guess there could be a world where it didn't error but did provide a different output and that would be useful for debugging? It seems like there isn't much need to add it in but since its in not worth taking out?